imports/lib/models/Base.ts

```diff
@@ -53,6 +49,14 @@ class Base<T extends BaseType> extends Mongo.Collection<T> {
     });
   }
 
+  // @ts-expect-error TS is correct that Partial<T> is not equivalent to
+  //   OptionalId<T>, but that's because OptionalId<T> isn't exactly correct.
+  //   Our SimpleSchemas will fill in some of the missing fields using
+  //   autoValues, but it's not possible to have a collection type that
+  //   distinguishes between fields that are present/required on read vs. on
+  //   write. The resulting declaration is looser than it should be, but it only
+  //   pushes validation from type checking to runtime (as SimpleSchema will
+  //   verify that required fields are actually present).
```

Comment:
OK. I think I understand the history here.

AIUI, the original override of `insert` to accept `Partial<T>` (#247) _did_ require `@ts-ignore`, as TypeScript couldn't tell that this is a valid subclassing of the `Mongo.Collection` parent class - i.e. it couldn't tell if `Partial<T>` was contravariant with `OptionalId<T>`, which is what the superclass takes. Arguably it's not, but that's mostly because we know better than the upstream type declarations that a `Partial<T>` is actually quasi-acceptable here.

When we originally introduced `ValidateShape` in #366, I believe that the `@ts-ignore` was made unnecessary. I'm not sure if it was actually unnecessary, or if TS was just unable to tell that anything untoward is going on. (It was probably confused by the ternaries). In either case, we didn't notice that it was unnecessary so we left it in.

Subsequently, when I turned on new linting configuration in #893, that config banned use of `@ts-ignore`. At the time, I was surprised to find that the declaration wasn't necessary, but it was easy enough to drop.

Now, with the new, simpler definition of `ValidateShape`, TS seems to have noticed what's going on again and is displeased. However, the new linting configuration is tolerant of `@ts-expect-error`, which has approximately the same effect as `@ts-ignore` but will error if there _is_ no type checking error. This should prevent us from missing if some future change again makes it unnecessary.